sympy/combinatorics/homomorphisms.py

```diff
@@ -199,29 +244,42 @@ def homomorphism(domain, codomain, gens, images=[]):
     gens.extend([g for g in generators if g not in gens])
     images = dict(zip(gens,images))
 
-    if not _check_homomorphism(domain, images, codomain.identity):
+    if check and not _check_homomorphism(domain, codomain, images):
         raise ValueError("The given images do not define a homomorphism")
     return GroupHomomorphism(domain, codomain, images)
 
-def _check_homomorphism(domain, images, identity):
+def _check_homomorphism(domain, codomain, images):
```

Comment:
For any domains, unless `check==False`. It depends on knowing the relators of the group so it would have to find an `FpGroup` presentation of a permutation group first.